docs/ref/modules/inventory-harvester/README.md

```diff
@@ -1,5 +1,71 @@
 # InventoryHarvester
-This module is responsible for receive inventory data and store it in the indexer. It is a simple module that receives a flatbuffer message and stores it in the indexer.
 
 ## Overview
 
+The **InventoryHarvester** module receives inventory data through flatbuffer messages and stores it in the Wazuh Indexer. By centralizing information about system inventory—such as hardware, software, network details, and File Integrity Monitoring (FIM) events—this module makes the data readily available for querying, visualization, and deeper analysis. Offloading this information to the Wazuh Indexer, rather than storing it on the Manager, helps prevent performance bottlenecks and data duplication.
+
+### Key Responsibilities
+
+- **Receive** inventory data (e.g., hardware/software/network details, FIM data) via flatbuffer messages.
+- **Index** all collected data in the Wazuh Indexer in accordance with global state requirements.
+- **Manage** bulk operations, reducing overhead and improving overall performance.
+- **Integrate** seamlessly with FIM and other Wazuh modules that rely on inventory data.
+
+### Data Flow
+
+1. **Message Ingestion**
+
+   - Flatbuffer messages arrive from the router IPC on the Manager side.
+   - Both the `Syscollector` modules and the `FIM` module contribute messages captured by the InventoryHarvester.
+
+2. **Deserialization & Validation**
+
+   - Each message is deserialized, validated, and prepared for indexing.
+
+3. **Indexer Connector**
+
+   - Validated data is batched and sent to the Wazuh Indexer in bulk, minimizing overhead.
+
+4. **Storage**
+
+   - The data is stored in a dedicated global state index, following Wazuh Common Schemas (WCS).
```

Comment:
We can add something like "there is one index for each type of ..."

Reply:
Done!